### PR TITLE
[libuv/librsync/libqrencode] Update to latest release version

### LINKED ISSUE #19894: [libuv] update to 1.42.0
Library name: libuv

New version number: 1.42.0

Other information that may be useful (release notes, etc...)
https://github.com/libuv/libuv/releases

### LINKED ISSUE #19937: [libqrencode] update to 4.1.1
Library name: libqrencode

New version number: 4.1.1

Other information that may be useful (release notes, etc...)
https://github.com/fukuchi/libqrencode/releases

### LINKED ISSUE #19939: [librsync] update to 2.3.2
Library name: librsync

New version number: 2.3.2

Other information that may be useful (release notes, etc...)
https://github.com/librsync/librsync/releases

### PR DESCRIPTION
**Pull request by PhoebeHui**
Fixes https://github.com/microsoft/vcpkg/issues/19937
Fixes https://github.com/microsoft/vcpkg/issues/19939
Fixes https://github.com/microsoft/vcpkg/issues/19894

Update libuv/librsync/libqrencode to latest release version

### CODE REVIEW

File: ports/libqrencode/portfile.cmake [hunk 1]
```diff
@@ -1,22 +1,20 @@
...
 )
 
-if("tool" IN_LIST FEATURES)
-    set(WITH_TOOLS YES)
-else()
-    set(WITH_TOOLS NO)
-endif()
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+    FEATURES
+      tool WITH_TOOLS
```
Comment (NancyLi1013) [inline ports/libqrencode/portfile.cmake:11]:
```suggestion
    FEATURES
        tool WITH_TOOLS
```

File: ports/libqrencode/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "libqrencode",
+  "version-string": "4.1.1",
```
Comment (NancyLi1013) [inline ports/libqrencode/vcpkg.json:3]:
```suggestion
  "version-semver": "4.1.1",
```

File: ports/librsync/portfile.cmake [hunk 3]
```diff
@@ -5,32 +5,30 @@ vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
...
 endif()
-if(EXISTS ${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll)
-    file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/debug/bin)
-    file(RENAME ${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll ${CURRENT_PACKAGES_DIR}/debug/bin/rsync.dll)
+if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll")
+    file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/debug/bin")
+    file(RENAME "${CURRENT_PACKAGES_DIR}/debug/lib/rsync.dll" "${CURRENT_PACKAGES_DIR}/debug/bin/rsync.dll")
 endif()
 
 if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
```
Comment (NancyLi1013) [inline ports/librsync/portfile.cmake:34]: `vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/include/librsync_export.h"`

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the updates!

**Pull request merged**